CIP-0062/README.md

```diff
@@ -0,0 +1,192 @@
+
+`UnknownChoiceError` - If the `choice` is not within the `proposal` option set.
+
+`InvalidBlockDateError` - If the `validUntil` is not a valid block date.
+
+`InvalidVotePlanError` - If the `votePlanId` is not a valid vote plan.
+
+`InvalidVoteOptionError` - If the `index` is not a valid vote option.
+
+# **`Delegation API`**
+
+## **api.getVotingKey**(): Promise\<cbor<PublicKey\>>
+
+Should return the in-use voting public-key. The wallet should maintain a reference to the current `adress_index` counter and return the public key for that index.
+
+### **Returns**
+cbor hex encoded representation of the public key.
+
+## **api.submitDelegation(delegation: Delegation): Promise\<SignedDelegationMetadata>**
+
+This endpoint should construct the cbor encoded delegation certificate according to the specs in [CIP-36 Example](https://github.com/Zeegomo/CIPs/blob/472181b9c69feeedae0b5b2db8b42d0cf4eb1a11/CIP-0036/README.md#example).
+
+It should then sign the certificate with the staking key as described in the same example as above.
+
+The implementation of this endpoint can make use of the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.signData` and `api.submitTx` to perform the broadcasting of the transaction containing the metadata. 
+
+Upon submission of the transaction containing the delegation cert as part of metadata, the wallet should store the delegation in its local storage and return an object that contains the delegation cert, the signature and the txhash of the transaction that the certificate was submitted with.
+
+## **`Delegation`**
+
+```
+export interface Delegation {
+    delegations: GovernanceKey[],
+    purpose: Purpose
+}
+```
+
+Defines the structure to be crafted and signed for delegation of voting & their respectively voting power.   Embeds the stake key and reward address from the wallet, and constructs a suitable nonce.
+
+***`delegations`***: List of keys and their voting weight to delegate voting power to.
+
+This should be a call that implicitly cbor encodes the `delegation` object and uses the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.submitTx` to submit the transaction. The resulting transaction hash should be returned.
+
+This should be trigger a request to the wallet to approve the transaction.
+
+### **Returns**
+
+```
+interface SignedDelegationMetadata {
+  certificate: DelegatedCertificate,
+  signature: string, // signature on the CIP-36 delegation certificate
+  txHash: string // of the transaction that submitted the delegation 
+}
+```
+
+#### **DelegatedCertificate**
+```
+interface DelegatedCertificate {
+  delegations: GovernanceKey[] // key delegations,
+  stakingPub: string // staking public key
+  rewardAddress: string // reward address
+  nonce: number //nonce used in the transaction
+  purpose: Purpose
+}
+```
+
+Errors: `APIError`, `TxSendError`
+
+## **Delegation Cert process**
```

Comment:
I would expect a similar breakdown for the Voting process, it would make the motivation behind the available calls clearer to the reader

Reply:
Will do.